feature-libs/order/components/order-details/order-document-flow/order-document-flow-dialog/order-document-flow-dialog.component.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * SPDX-FileCopyrightText: 2025 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  DestroyRef,
+  ElementRef,
+  inject,
+  signal,
+  ViewChild,
+} from '@angular/core';
+import {
+  GlobalMessageType,
+  TranslationService,
+  isNotNullable,
+} from '@spartacus/core';
+import {
+  FocusConfig,
+  ICON_TYPE,
+  LaunchDialogService,
+} from '@spartacus/storefront';
+import { BehaviorSubject, Observable, of } from 'rxjs';
+import {
+  catchError,
+  filter,
+  map,
+  shareReplay,
+  switchMap,
+  tap,
+  withLatestFrom,
+} from 'rxjs/operators';
+import {
+  OrderDocumentFlowFacade,
+  SapOrderSubsequentDocument,
+  SapOrderSubsequentDocumentEntry,
+} from '@spartacus/order/root';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+@Component({
+  selector: 'cx-order-document-flow-dialog',
+  templateUrl: './order-document-flow-dialog.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  standalone: false,
+})
+export class OrderDocumentFlowDialogComponent {
+  protected launchDialogService = inject(LaunchDialogService);
+  protected orderDocumentFlowFacade = inject(OrderDocumentFlowFacade);
+  protected translation = inject(TranslationService);
+  protected cd = inject(ChangeDetectorRef);
+  protected destroyRef = inject(DestroyRef);
+  @ViewChild('scrollContainer') scrollContainerRef: ElementRef;
+  protected savedScrollPosition = 0;
+
+  globalMessageType = GlobalMessageType;
+  iconTypes = ICON_TYPE;
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: true,
+    focusOnEscape: true,
+  };
+
+  displayDocumentEntries = signal(false);
+
+  orderCode$: Observable<string> = this.launchDialogService.data$.pipe(
```

Comment:
I don't see usages in the template
```suggestion
  protected orderCode$: Observable<string> = this.launchDialogService.data$.pipe(
```

Reply:
added protected

Reply:
I had to update unit test for that component after that change